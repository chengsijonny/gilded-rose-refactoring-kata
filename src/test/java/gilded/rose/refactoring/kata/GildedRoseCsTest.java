package gilded.rose.refactoring.kata;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

public class GildedRoseCsTest {

    @Test
    public void updateQuality() {
        //        Approvals.verify(itemString);
        CombinationApprovals.verifyAllCombinations(
                this::doUpdateQuality,
                new String[]{"+5 Dexterity Vest", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert",
                },
                new Integer[]{-1, 0, 1, 4, 5, 11},
                new Integer[]{0, 1, 2, 3, 20, 46, 47, 48, 49, 50}
        );
    }

    private String doUpdateQuality(String name, int sellIn, int quality) {
        ItemCs[] items = new ItemCs[]{
                new ItemCs(name, sellIn, quality)};


        GildedRoseCs app = new GildedRoseCs(items);
        app.updateQuality();
        return app.items[0].toString();
    }
}
