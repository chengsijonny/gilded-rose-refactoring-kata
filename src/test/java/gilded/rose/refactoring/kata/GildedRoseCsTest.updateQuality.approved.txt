[+5 Dexterity Vest, -1, 0] => +5 Dexterity Vest, -2, 0 
[+5 Dexterity Vest, -1, 1] => +5 Dexterity Vest, -2, 0 
[+5 Dexterity Vest, -1, 2] => +5 Dexterity Vest, -2, 0 
[+5 Dexterity Vest, -1, 3] => +5 Dexterity Vest, -2, 1 
[+5 Dexterity Vest, -1, 20] => +5 Dexterity Vest, -2, 18 
[+5 Dexterity Vest, -1, 46] => +5 Dexterity Vest, -2, 44 
[+5 Dexterity Vest, -1, 47] => +5 Dexterity Vest, -2, 45 
[+5 Dexterity Vest, -1, 48] => +5 Dexterity Vest, -2, 46 
[+5 Dexterity Vest, -1, 49] => +5 Dexterity Vest, -2, 47 
[+5 Dexterity Vest, -1, 50] => +5 Dexterity Vest, -2, 48 
[+5 Dexterity Vest, 0, 0] => +5 Dexterity Vest, -1, 0 
[+5 Dexterity Vest, 0, 1] => +5 Dexterity Vest, -1, 0 
[+5 Dexterity Vest, 0, 2] => +5 Dexterity Vest, -1, 0 
[+5 Dexterity Vest, 0, 3] => +5 Dexterity Vest, -1, 1 
[+5 Dexterity Vest, 0, 20] => +5 Dexterity Vest, -1, 18 
[+5 Dexterity Vest, 0, 46] => +5 Dexterity Vest, -1, 44 
[+5 Dexterity Vest, 0, 47] => +5 Dexterity Vest, -1, 45 
[+5 Dexterity Vest, 0, 48] => +5 Dexterity Vest, -1, 46 
[+5 Dexterity Vest, 0, 49] => +5 Dexterity Vest, -1, 47 
[+5 Dexterity Vest, 0, 50] => +5 Dexterity Vest, -1, 48 
[+5 Dexterity Vest, 1, 0] => +5 Dexterity Vest, 0, 0 
[+5 Dexterity Vest, 1, 1] => +5 Dexterity Vest, 0, 0 
[+5 Dexterity Vest, 1, 2] => +5 Dexterity Vest, 0, 1 
[+5 Dexterity Vest, 1, 3] => +5 Dexterity Vest, 0, 2 
[+5 Dexterity Vest, 1, 20] => +5 Dexterity Vest, 0, 19 
[+5 Dexterity Vest, 1, 46] => +5 Dexterity Vest, 0, 45 
[+5 Dexterity Vest, 1, 47] => +5 Dexterity Vest, 0, 46 
[+5 Dexterity Vest, 1, 48] => +5 Dexterity Vest, 0, 47 
[+5 Dexterity Vest, 1, 49] => +5 Dexterity Vest, 0, 48 
[+5 Dexterity Vest, 1, 50] => +5 Dexterity Vest, 0, 49 
[+5 Dexterity Vest, 4, 0] => +5 Dexterity Vest, 3, 0 
[+5 Dexterity Vest, 4, 1] => +5 Dexterity Vest, 3, 0 
[+5 Dexterity Vest, 4, 2] => +5 Dexterity Vest, 3, 1 
[+5 Dexterity Vest, 4, 3] => +5 Dexterity Vest, 3, 2 
[+5 Dexterity Vest, 4, 20] => +5 Dexterity Vest, 3, 19 
[+5 Dexterity Vest, 4, 46] => +5 Dexterity Vest, 3, 45 
[+5 Dexterity Vest, 4, 47] => +5 Dexterity Vest, 3, 46 
[+5 Dexterity Vest, 4, 48] => +5 Dexterity Vest, 3, 47 
[+5 Dexterity Vest, 4, 49] => +5 Dexterity Vest, 3, 48 
[+5 Dexterity Vest, 4, 50] => +5 Dexterity Vest, 3, 49 
[+5 Dexterity Vest, 5, 0] => +5 Dexterity Vest, 4, 0 
[+5 Dexterity Vest, 5, 1] => +5 Dexterity Vest, 4, 0 
[+5 Dexterity Vest, 5, 2] => +5 Dexterity Vest, 4, 1 
[+5 Dexterity Vest, 5, 3] => +5 Dexterity Vest, 4, 2 
[+5 Dexterity Vest, 5, 20] => +5 Dexterity Vest, 4, 19 
[+5 Dexterity Vest, 5, 46] => +5 Dexterity Vest, 4, 45 
[+5 Dexterity Vest, 5, 47] => +5 Dexterity Vest, 4, 46 
[+5 Dexterity Vest, 5, 48] => +5 Dexterity Vest, 4, 47 
[+5 Dexterity Vest, 5, 49] => +5 Dexterity Vest, 4, 48 
[+5 Dexterity Vest, 5, 50] => +5 Dexterity Vest, 4, 49 
[+5 Dexterity Vest, 11, 0] => +5 Dexterity Vest, 10, 0 
[+5 Dexterity Vest, 11, 1] => +5 Dexterity Vest, 10, 0 
[+5 Dexterity Vest, 11, 2] => +5 Dexterity Vest, 10, 1 
[+5 Dexterity Vest, 11, 3] => +5 Dexterity Vest, 10, 2 
[+5 Dexterity Vest, 11, 20] => +5 Dexterity Vest, 10, 19 
[+5 Dexterity Vest, 11, 46] => +5 Dexterity Vest, 10, 45 
[+5 Dexterity Vest, 11, 47] => +5 Dexterity Vest, 10, 46 
[+5 Dexterity Vest, 11, 48] => +5 Dexterity Vest, 10, 47 
[+5 Dexterity Vest, 11, 49] => +5 Dexterity Vest, 10, 48 
[+5 Dexterity Vest, 11, 50] => +5 Dexterity Vest, 10, 49 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 1] => Aged Brie, -2, 3 
[Aged Brie, -1, 2] => Aged Brie, -2, 4 
[Aged Brie, -1, 3] => Aged Brie, -2, 5 
[Aged Brie, -1, 20] => Aged Brie, -2, 22 
[Aged Brie, -1, 46] => Aged Brie, -2, 48 
[Aged Brie, -1, 47] => Aged Brie, -2, 49 
[Aged Brie, -1, 48] => Aged Brie, -2, 50 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 1] => Aged Brie, -1, 3 
[Aged Brie, 0, 2] => Aged Brie, -1, 4 
[Aged Brie, 0, 3] => Aged Brie, -1, 5 
[Aged Brie, 0, 20] => Aged Brie, -1, 22 
[Aged Brie, 0, 46] => Aged Brie, -1, 48 
[Aged Brie, 0, 47] => Aged Brie, -1, 49 
[Aged Brie, 0, 48] => Aged Brie, -1, 50 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, 1, 0] => Aged Brie, 0, 1 
[Aged Brie, 1, 1] => Aged Brie, 0, 2 
[Aged Brie, 1, 2] => Aged Brie, 0, 3 
[Aged Brie, 1, 3] => Aged Brie, 0, 4 
[Aged Brie, 1, 20] => Aged Brie, 0, 21 
[Aged Brie, 1, 46] => Aged Brie, 0, 47 
[Aged Brie, 1, 47] => Aged Brie, 0, 48 
[Aged Brie, 1, 48] => Aged Brie, 0, 49 
[Aged Brie, 1, 49] => Aged Brie, 0, 50 
[Aged Brie, 1, 50] => Aged Brie, 0, 50 
[Aged Brie, 4, 0] => Aged Brie, 3, 1 
[Aged Brie, 4, 1] => Aged Brie, 3, 2 
[Aged Brie, 4, 2] => Aged Brie, 3, 3 
[Aged Brie, 4, 3] => Aged Brie, 3, 4 
[Aged Brie, 4, 20] => Aged Brie, 3, 21 
[Aged Brie, 4, 46] => Aged Brie, 3, 47 
[Aged Brie, 4, 47] => Aged Brie, 3, 48 
[Aged Brie, 4, 48] => Aged Brie, 3, 49 
[Aged Brie, 4, 49] => Aged Brie, 3, 50 
[Aged Brie, 4, 50] => Aged Brie, 3, 50 
[Aged Brie, 5, 0] => Aged Brie, 4, 1 
[Aged Brie, 5, 1] => Aged Brie, 4, 2 
[Aged Brie, 5, 2] => Aged Brie, 4, 3 
[Aged Brie, 5, 3] => Aged Brie, 4, 4 
[Aged Brie, 5, 20] => Aged Brie, 4, 21 
[Aged Brie, 5, 46] => Aged Brie, 4, 47 
[Aged Brie, 5, 47] => Aged Brie, 4, 48 
[Aged Brie, 5, 48] => Aged Brie, 4, 49 
[Aged Brie, 5, 49] => Aged Brie, 4, 50 
[Aged Brie, 5, 50] => Aged Brie, 4, 50 
[Aged Brie, 11, 0] => Aged Brie, 10, 1 
[Aged Brie, 11, 1] => Aged Brie, 10, 2 
[Aged Brie, 11, 2] => Aged Brie, 10, 3 
[Aged Brie, 11, 3] => Aged Brie, 10, 4 
[Aged Brie, 11, 20] => Aged Brie, 10, 21 
[Aged Brie, 11, 46] => Aged Brie, 10, 47 
[Aged Brie, 11, 47] => Aged Brie, 10, 48 
[Aged Brie, 11, 48] => Aged Brie, 10, 49 
[Aged Brie, 11, 49] => Aged Brie, 10, 50 
[Aged Brie, 11, 50] => Aged Brie, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 2] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 3] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 20] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 46] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 47] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 48] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 2] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 3] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 20] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 46] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 47] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 48] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 1, 0] => Backstage passes to a TAFKAL80ETC concert, 0, 3 
[Backstage passes to a TAFKAL80ETC concert, 1, 1] => Backstage passes to a TAFKAL80ETC concert, 0, 4 
[Backstage passes to a TAFKAL80ETC concert, 1, 2] => Backstage passes to a TAFKAL80ETC concert, 0, 5 
[Backstage passes to a TAFKAL80ETC concert, 1, 3] => Backstage passes to a TAFKAL80ETC concert, 0, 6 
[Backstage passes to a TAFKAL80ETC concert, 1, 20] => Backstage passes to a TAFKAL80ETC concert, 0, 23 
[Backstage passes to a TAFKAL80ETC concert, 1, 46] => Backstage passes to a TAFKAL80ETC concert, 0, 49 
[Backstage passes to a TAFKAL80ETC concert, 1, 47] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 48] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 49] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 50] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 4, 0] => Backstage passes to a TAFKAL80ETC concert, 3, 3 
[Backstage passes to a TAFKAL80ETC concert, 4, 1] => Backstage passes to a TAFKAL80ETC concert, 3, 4 
[Backstage passes to a TAFKAL80ETC concert, 4, 2] => Backstage passes to a TAFKAL80ETC concert, 3, 5 
[Backstage passes to a TAFKAL80ETC concert, 4, 3] => Backstage passes to a TAFKAL80ETC concert, 3, 6 
[Backstage passes to a TAFKAL80ETC concert, 4, 20] => Backstage passes to a TAFKAL80ETC concert, 3, 23 
[Backstage passes to a TAFKAL80ETC concert, 4, 46] => Backstage passes to a TAFKAL80ETC concert, 3, 49 
[Backstage passes to a TAFKAL80ETC concert, 4, 47] => Backstage passes to a TAFKAL80ETC concert, 3, 50 
[Backstage passes to a TAFKAL80ETC concert, 4, 48] => Backstage passes to a TAFKAL80ETC concert, 3, 50 
[Backstage passes to a TAFKAL80ETC concert, 4, 49] => Backstage passes to a TAFKAL80ETC concert, 3, 50 
[Backstage passes to a TAFKAL80ETC concert, 4, 50] => Backstage passes to a TAFKAL80ETC concert, 3, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] => Backstage passes to a TAFKAL80ETC concert, 4, 4 
[Backstage passes to a TAFKAL80ETC concert, 5, 2] => Backstage passes to a TAFKAL80ETC concert, 4, 5 
[Backstage passes to a TAFKAL80ETC concert, 5, 3] => Backstage passes to a TAFKAL80ETC concert, 4, 6 
[Backstage passes to a TAFKAL80ETC concert, 5, 20] => Backstage passes to a TAFKAL80ETC concert, 4, 23 
[Backstage passes to a TAFKAL80ETC concert, 5, 46] => Backstage passes to a TAFKAL80ETC concert, 4, 49 
[Backstage passes to a TAFKAL80ETC concert, 5, 47] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 48] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => Backstage passes to a TAFKAL80ETC concert, 10, 2 
[Backstage passes to a TAFKAL80ETC concert, 11, 2] => Backstage passes to a TAFKAL80ETC concert, 10, 3 
[Backstage passes to a TAFKAL80ETC concert, 11, 3] => Backstage passes to a TAFKAL80ETC concert, 10, 4 
[Backstage passes to a TAFKAL80ETC concert, 11, 20] => Backstage passes to a TAFKAL80ETC concert, 10, 21 
[Backstage passes to a TAFKAL80ETC concert, 11, 46] => Backstage passes to a TAFKAL80ETC concert, 10, 47 
[Backstage passes to a TAFKAL80ETC concert, 11, 47] => Backstage passes to a TAFKAL80ETC concert, 10, 48 
[Backstage passes to a TAFKAL80ETC concert, 11, 48] => Backstage passes to a TAFKAL80ETC concert, 10, 49 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => Backstage passes to a TAFKAL80ETC concert, 10, 50 
